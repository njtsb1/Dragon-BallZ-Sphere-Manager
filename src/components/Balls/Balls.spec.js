import React from 'react'
import { render, fireEvent, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Balls from './Balls'

//mocks
import { profile } from '../../mocks/profile.json';
import profileSuccess from '../../mocks/profileSuccess.json';
import spheres from '../../mocks/spheres.json';
import spheresSuccess from '../../mocks/spheresSuccess.json';

it('Should render all dragon balls', () => {
  const { container } = render(<Balls balls={spheres.balls} profile={profile} />)
  expect(container).toMatchSnapshot()
})

it('Should show not found balls label and button', () => {
  const { getAllByText } = render(<Balls balls={spheres.balls} profile={profile} />)

  const notFound = !!(getAllByText('Not found').length > 0)
  const validateButton = !!(getAllByText('found').length > 0)

  expect(notFound).toBeTruthy()
  expect(validateButton).toBeTruthy()
})

it('Should open modal to validate', () => {
  const { debug, getAllByText, getByText, container } = render(<Balls balls={spheres.balls} profile={profile} />)
  const validateButton = getAllByText('I found')

  fireEvent.click(validateButton[0])

  expect(getByText('Validate')).toBeTruthy()
  expect(container).toMatchSnapshot()

})

it('Should filter', async () => {
  const { debug, getByTestId, getByText, getAllByText, container } = render(<Balls balls={spheres.balls} profile={profile} />)
  const filter = getByTestId('filter')
  fireEvent.click(filter)

  const myDragonBalls = getByText('I dont have').closest('option')
  fireEvent.click(myDragonBalls)

  await waitFor(() => {
    expect(getAllByText('Not found').length).toBe(3)
    // console.log(getAllByText('Encontrada').length)
    // expect(getAllByText('Encontrada')).toBe(3).not.toBeInTheDocument()
  })
})
