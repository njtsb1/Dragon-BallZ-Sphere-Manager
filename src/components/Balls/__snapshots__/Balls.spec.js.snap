// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should open modal to validate 1`] = `
<div>
  <div
    class="css-mv96ze"
  >
    <div
      class="css-1thq8i2"
    >
      Spheres
    </div>
    <div
      class="form-group"
    >
      <label
        class=""
        for="filter"
      >
        Filter
      </label>
      <select
        class="form-control"
        data-testid="filter"
        id="filter"
        name="select"
      >
        <option
          value="all"
        >
          All spheres
        </option>
        <option
          value="me"
        >
          My spheres
        </option>
        <option
          value="notme"
        >
          I don't have
        </option>
      </select>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-sm-3"
    >
      <div
        class="sc-AxjAm iGDTZF card"
      >
        <img
          alt="1 star"
          class="card-img-top"
          src="https://i.pinimg.com/564x/a5/f9/a2/a5f9a2eb5c0bfb1f66988696e1f31334.jpg"
          width="100%"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title"
          >
            1 star
          </div>
          <span
            class="badge badge-success"
          >
            Found
          </span>
        </div>
      </div>
    </div>
    <div
      class="col-sm-3"
    >
      <div
        class="sc-AxjAm iGDTZF card"
      >
        <img
          alt="2 stars"
          class="card-img-top"
          src="https://i.pinimg.com/564x/0b/72/75/0b72755a5ea5e1999d6fbfd3aa544355.jpg"
          width="100%"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title"
          >
            2 stars
          </div>
          <span
            class="badge badge-danger"
          >
            Not found
          </span>
          <button
            class="btn btn-warning btn-sm"
            type="button"
          >
            I found
          </button>
        </div>
      </div>
    </div>
    <div
      class="col-sm-3"
    >
      <div
        class="sc-AxjAm iGDTZF card"
      >
        <img
          alt="3 stars"
          class="card-img-top"
          src="https://i.pinimg.com/564x/32/eb/b8/32ebb8536d0fa4108eef460694586d39.jpg"
          width="100%"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title"
          >
            3 stars
          </div>
          <span
            class="badge badge-danger"
          >
            Not found
          </span>
          <button
            class="btn btn-warning btn-sm"
            type="button"
          >
            I found
          </button>
        </div>
      </div>
    </div>
    <div
      class="col-sm-3"
    >
      <div
        class="sc-AxjAm iGDTZF card"
      >
        <img
          alt="4 stars"
          class="card-img-top"
          src="https://i.pinimg.com/564x/7a/7f/76/7a7f76ea01702078582a1ae8b7344fb3.jpg"
          width="100%"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title"
          >
            4 stars
          </div>
          <span
            class="badge badge-danger"
          >
            Not found
          </span>
          <button
            class="btn btn-warning btn-sm"
            type="button"
          >
            I found
          </button>
        </div>
      </div>
    </div>
    <div
      class="col-sm-3"
    >
      <div
        class="sc-AxjAm iGDTZF card"
      >
        <img
          alt="5 stars"
          class="card-img-top"
          src="https://i.pinimg.com/564x/13/19/62/1319628426fa9d3783705daf7db525f8.jpg"
          width="100%"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title"
          >
            5 stars
          </div>
          <span
            class="badge badge-success"
          >
            Found
          </span>
        </div>
      </div>
    </div>
    <div
      class="col-sm-3"
    >
      <div
        class="sc-AxjAm iGDTZF card"
      >
        <img
          alt="6 stars"
          class="card-img-top"
          src="https://i.pinimg.com/originals/6c/72/47/6c7247dfb67e18add93d682dc9fdabcc.png"
          width="100%"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title"
          >
            6 stars
          </div>
          <span
            class="badge badge-success"
          >
            Found
          </span>
        </div>
      </div>
    </div>
    <div
      class="col-sm-3"
    >
      <div
        class="sc-AxjAm iGDTZF card"
      >
        <img
          alt="7 stars"
          class="card-img-top"
          src="https://i.pinimg.com/564x/ec/8e/c5/ec8ec55eb1b4d4dfa3f7b2cbffe5bc6c.jpg"
          width="100%"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title"
          >
            7 stars
          </div>
          <span
            class="badge badge-success"
          >
            Found
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Should render all dragon balls 1`] = `
<div>
  <div
    class="css-mv96ze"
  >
    <div
      class="css-1thq8i2"
    >
      Esferas
    </div>
    <div
      class="form-group"
    >
      <label
        class=""
        for="filter"
      >
        Filter
      </label>
      <select
        class="form-control"
        data-testid="filter"
        id="filter"
        name="select"
      >
        <option
          value="all"
        >
          Todas as esferas
        </option>
        <option
          value="me"
        >
          Minhas esferas
        </option>
        <option
          value="notme"
        >
          NÃ£o tenho
        </option>
      </select>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-sm-3"
    >
      <div
        class="sc-AxjAm iGDTZF card"
      >
        <img
          alt="1 star"
          class="card-img-top"
          src="https://i.pinimg.com/564x/a5/f9/a2/a5f9a2eb5c0bfb1f66988696e1f31334.jpg"
          width="100%"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title"
          >
            1 star
          </div>
          <span
            class="badge badge-success"
          >
            Found
          </span>
        </div>
      </div>
    </div>
    <div
      class="col-sm-3"
    >
      <div
        class="sc-AxjAm iGDTZF card"
      >
        <img
          alt="2 stars"
          class="card-img-top"
          src="https://i.pinimg.com/564x/0b/72/75/0b72755a5ea5e1999d6fbfd3aa544355.jpg"
          width="100%"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title"
          >
            2 stars
          </div>
          <span
            class="badge badge-danger"
          >
            Not found
          </span>
          <button
            class="btn btn-warning btn-sm"
            type="button"
          >
            I found
          </button>
        </div>
      </div>
    </div>
    <div
      class="col-sm-3"
    >
      <div
        class="sc-AxjAm iGDTZF card"
      >
        <img
          alt="3 stars"
          class="card-img-top"
          src="https://i.pinimg.com/564x/32/eb/b8/32ebb8536d0fa4108eef460694586d39.jpg"
          width="100%"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title"
          >
            3 stars
          </div>
          <span
            class="badge badge-danger"
          >
            Not found
          </span>
          <button
            class="btn btn-warning btn-sm"
            type="button"
          >
            I found
          </button>
        </div>
      </div>
    </div>
    <div
      class="col-sm-3"
    >
      <div
        class="sc-AxjAm iGDTZF card"
      >
        <img
          alt="4 stars"
          class="card-img-top"
          src="https://i.pinimg.com/564x/7a/7f/76/7a7f76ea01702078582a1ae8b7344fb3.jpg"
          width="100%"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title"
          >
            4 stars
          </div>
          <span
            class="badge badge-danger"
          >
            Not found
          </span>
          <button
            class="btn btn-warning btn-sm"
            type="button"
          >
            I found
          </button>
        </div>
      </div>
    </div>
    <div
      class="col-sm-3"
    >
      <div
        class="sc-AxjAm iGDTZF card"
      >
        <img
          alt="5 stars"
          class="card-img-top"
          src="https://i.pinimg.com/564x/13/19/62/1319628426fa9d3783705daf7db525f8.jpg"
          width="100%"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title"
          >
            5 stars
          </div>
          <span
            class="badge badge-success"
          >
            Found
          </span>
        </div>
      </div>
    </div>
    <div
      class="col-sm-3"
    >
      <div
        class="sc-AxjAm iGDTZF card"
      >
        <img
          alt="6 stars"
          class="card-img-top"
          src="https://i.pinimg.com/originals/6c/72/47/6c7247dfb67e18add93d682dc9fdabcc.png"
          width="100%"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title"
          >
            6 stars
          </div>
          <span
            class="badge badge-success"
          >
            Found
          </span>
        </div>
      </div>
    </div>
    <div
      class="col-sm-3"
    >
      <div
        class="sc-AxjAm iGDTZF card"
      >
        <img
          alt="7 stars"
          class="card-img-top"
          src="https://i.pinimg.com/564x/ec/8e/c5/ec8ec55eb1b4d4dfa3f7b2cbffe5bc6c.jpg"
          width="100%"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title"
          >
            7 stars
          </div>
          <span
            class="badge badge-success"
          >
            Found
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
